packages/contracts/contracts/Unirep.sol

```diff
@@ -378,47 +398,61 @@ contract Unirep is IUnirep, zkSNARKHelper, Hasher, VerifySignature {
 
     /**
      * A user spend reputation via an attester, the non-zero nullifiers will be processed as a negative attestation
-     * @param input The epoch key and its proof and the public signals
+     * publicSignals[0: maxReputationBudget ] = [ reputationNullifiers ]
+     * publicSignals[maxReputationBudget    ] = [ epoch ]
+     * publicSignals[maxReputationBudget + 1] = [ epochKey ]
+     * publicSignals[maxReputationBudget + 2] = [ globalStateTree ]
+     * publicSignals[maxReputationBudget + 3] = [ attesterId ]
+     * publicSignals[maxReputationBudget + 4] = [ proveReputationAmount ]
+     * publicSignals[maxReputationBudget + 5] = [ minRep ]
+     * publicSignals[maxReputationBudget + 6] = [ proveGraffiti ]
+     * publicSignals[maxReputationBudget + 7] = [ graffitiPreImage ]
+     * @param publicSignals The public signals of the reputation proof
+     * @param proof The The proof of the reputation proof
      */
-    function spendReputation(ReputationProof memory input) external payable {
-        bytes32 proofNullifier = Hasher.hashReputationProof(input);
+    function spendReputation(
+        uint256[] memory publicSignals,
+        uint256[8] memory proof
+    ) external payable {
```

Comment:
Why we add payable here? Do we handle any payment transaction?

Reply:
yes there is an attestation submitted
and we check `attestingFee`

Reply:
Oh! I see. Thanks